test/e2e/actions/cloud/azure.go

```diff
@@ -392,6 +399,62 @@ func (a *AzureAction) deletePrivateEndpoint(ctx context.Context, endpointName st
 	return nil
 }
 
+func (a *AzureAction) CreateKeyVault(keyName string) (string, error) {
+	a.t.Helper()
+
+	ctx := context.Background()
+
+	client, err := azkeys.NewClient(AzureKeyVaultURI, a.credentials, nil)
+	if err != nil {
+		return "", err
+	}
+
+	params := azkeys.CreateKeyParameters{
+		KeySize: toptr.MakePtr[int32](2048),
+		Kty:     toptr.MakePtr(azkeys.KeyTypeRSA),
+	}
+
+	r, err := client.CreateKey(ctx, keyName, params, nil)
+	if err != nil {
+		return "", err
+	}
+
+	a.t.Cleanup(func() {
+		err = a.deleteKeyVault(ctx, keyName)
```

Comment:
We are not doing auto clean up for AWS right?
Can you explain why is that the case?

Reply:
I'm not sure, the AWS tests were already here. There is some setup done for AWS in the tests, but as you know we need to have already done some setup manually. Would be nice to eventually have everything happen in-test (infrastructure-wise) so tests don't need to be run specifically in our cloud accounts